src/lib.rs

```diff
@@ -165,12 +194,7 @@ pub extern "C" fn rustls_client_config_builder_load_native_roots(
     config: *mut rustls_client_config_builder,
 ) -> rustls_result {
     ffi_panic_boundary! {
-        let mut config: &mut ClientConfig = unsafe {
-            match (config as *mut ClientConfig).as_mut() {
-                Some(c) => c,
-                None => return rustls_result::NullParameter,
-            }
-        };
+        let mut config = checked_ptr_mut!(config, ClientConfig);
```

Comment:
nit: for consistency with other usages of `checked_ptr_ref` and `checked_ptr_mut`, and what I think is your preferred style, you might do `let config: &mut ClientConfig = checked_ptr_mut!(config, ClientConfig);`. On the other hand, that would repeat the type `ClientConfig` twice, so perhaps this usage is better.